pkg/ingester/ingester.go

```diff
@@ -373,10 +373,8 @@ func (i *Ingester) Push(ctx context.Context, req *client.WriteRequest) (*client.
 	}
 
 	if len(req.Metadata) > 0 {
-		// Given requests can only contain either metadata or samples, no-op if there is metadata for now.
```

Comment:
Technically, this is "no longer true" from an implementation perspective. The remote write protocol only sends metadata or samples (for now) but IMO that comment/responsibility belongs in the distributor. The ingester should operate under no assumption that you'd receive one or the other if we can (e.g. no increase of complexity). 

Obviously, this is my own personal opinion with what I know about the system, but others feel free to suggest any alternatives.

Reply:
If we're going to handle samples and metadata in the same push, then I would suggest that we return errors for them separately. As it is now, errors reported by ingesters will be counted for both samples and metadata (provided both were present in the request).

Reply:
Sorry Peter, I'm not following your comment. With this change, we're no longer making assumptions that a `WriteRequest` can come with one or the other.

The error handling in `send` is purely for metrics reporting. In the ingesters, we simply log when we receive metadata, so erroring out on that is not an option at the moment. 

The comment I originally left in the distributors is an operational concern, but for the code itself, it wouldn't make a difference - that is indented.